libs/x-content/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -329,28 +343,19 @@ public CharBuffer charBufferOrNull() throws IOException {
         return readList(parser, ORDERED_MAP_FACTORY);
     }
 
-    static Map<String, Object> readMap(XContentParser parser, MapFactory mapFactory) throws IOException {
-        Map<String, Object> map = mapFactory.newMap();
-        XContentParser.Token token = parser.currentToken();
-        if (token == null) {
-            token = parser.nextToken();
-        }
-        if (token == XContentParser.Token.START_OBJECT) {
-            token = parser.nextToken();
-        }
-        for (; token == XContentParser.Token.FIELD_NAME; token = parser.nextToken()) {
-            // Must point to field name
-            String fieldName = parser.currentName();
-            // And then the value...
-            token = parser.nextToken();
-            Object value = readValue(parser, mapFactory, token);
-            map.put(fieldName, value);
-        }
-        return map;
+    static Map<String, Object> readMap(XContentParser parser, MapFactory<Object> mapFactory) throws IOException {
+        return readGenericMap(parser, mapFactory, p -> readValue(p, mapFactory));
     }
 
-    static Map<String, String> readMapStrings(XContentParser parser, MapStringsFactory mapStringsFactory) throws IOException {
-        Map<String, String> map = mapStringsFactory.newMap();
+    static Map<String, String> readMapStrings(XContentParser parser, MapFactory<String> mapFactory) throws IOException {
+        return readGenericMap(parser, mapFactory, p -> p.text());
```

Comment:
can we replace `p -> p.text()` with `XContentParser::text`?

Reply:
Done.